### DISCUSSION TITLE
Rocket Redirect,  HTTP 303 on GET?

### DISCUSSION THREAD

**Discussion by kallisti5** [rwf2/Rocket#2417] [Questions]
We generally do an HTTP 302 redirect to forward users to s3... however I just noticed rocket is doing a HTTP 303 on a GET.  This seems... [incorrect](https://en.wikipedia.org/wiki/HTTP_303)?    HTTP 303 is only supposed to occur on a post.

I'm looking into this because we're requesting partials which is seemingly not making it "through" the http 303.

```
$ curl -r 0-25 https://eu.hpkg.haiku-os.org/haiku/master/x86_64/current/repo -v
   HTTP/2 303
   location: https:// ....
```

```
$ curl -r 0-99 https://eu.hpkg.haiku-os.org/haikuports/master/x86_64/current/repo -v
   > range: bytes=0-99
   data
```

Code: https://github.com/haiku/hpkgbouncer/blob/master/src/main.rs#L105

On get, do a Redirect.  Gets a 303

**Comment by wezm** [comment]
I think 303 is fine for GET requests:

> This status code is applicable to any HTTP method.
> A 303 response to a GET request indicates that the origin server does
>   not have a representation of the target resource that can be
>   transferred by the server over HTTP.  However, the Location field
>   value refers to a resource that is descriptive of the target
>   resource

https://www.rfc-editor.org/rfc/rfc7231#section-6.4.4

**Comment by kallisti5** [comment]
Ah ok. That wikipedia article is incorrect then.
Maybe the Rocket Redirect response needs to pass the range header back to the client with the HTTP 303 redirect :thinking:

**Reply by wezm** [reply]
> Ah ok. That wikipedia article is incorrect then. 

The second paragraph included that text about GET.

> Maybe the Rocket Redirect response needs to pass the range header back to the client with the HTTP 303 redirect

`Range` is a client side (request) header so I don't think Rocket can do that. 

`curl` is not following the redirect in your command above, it's just making the request and printing the 303 response. If you want it to follow redirects you have to supply `-L`. If you do that you will see it makes a second request to the redirected url with the `Range` header present.
